testing/e2e/e2e_api_test.go

```diff
@@ -205,3 +253,271 @@ func (s *BeaconKitE2ESuite) TestValidatorsWithSpecificStatus() {
 		s.Require().Equal(apiv1.ValidatorStateActiveOngoing, validator.Status)
 	}
 }
+
+// TestValidatorBalances tests querying validator balances.
+func (s *BeaconKitE2ESuite) TestValidatorBalances() {
+	client := s.initBeaconTest()
+
+	balancesResp, err := client.ValidatorBalances(s.Ctx(), &beaconapi.ValidatorBalancesOpts{
+		State: utils.StateIDHead,
+	})
+	s.Require().NoError(err)
+	s.Require().NotNil(balancesResp)
+
+	// Verify the response is not empty
+	s.Require().NotNil(balancesResp.Data)
+	s.Require().NotEmpty(balancesResp.Data)
+
+	balanceMap := balancesResp.Data
+	for _, balance := range balanceMap {
+		s.Require().True(balance > 0, "Validator balance should be positive")
+		// 4e12 Gwei = 4 * 10^12 Gwei = 4,000,000,000,000 Gwei = 4000 BERA
+		s.Require().True(balance <= 4e12, "Validator balance should not exceed 4e12 gwei (4000 BERA)")
+	}
+}
+
+// TestValidatorBalancesWithSpecificIndices tests querying validator balances with specific indices.
+func (s *BeaconKitE2ESuite) TestValidatorBalancesWithSpecificIndices() {
+	client := s.initBeaconTest()
+
+	indices := []phase0.ValidatorIndex{0}
+
+	balancesResp, err := client.ValidatorBalances(s.Ctx(), &beaconapi.ValidatorBalancesOpts{
+		State:   utils.StateIDHead,
+		Indices: indices,
+	})
+	s.Require().NoError(err)
+	s.Require().NotNil(balancesResp)
+
+	// Verify the response is not empty
+	s.Require().NotNil(balancesResp.Data)
+	s.Require().Len(balancesResp.Data, len(indices))
+	s.Require().NotEmpty(balancesResp.Data)
+
+	// Verify balance data
+	for index, balance := range balancesResp.Data {
+		s.Require().NotNil(balance)
+		s.Require().Contains(indices, index)
+		s.Require().True(balance > 0, "Validator balance should be positive")
+		// 4e12 Gwei = 4 * 10^12 Gwei = 4,000,000,000,000 Gwei = 4000 BERA
+		s.Require().True(balance <= 4e12, "Validator balance should not exceed 4000 BERA")
+	}
+}
+
+// TestValidatorBalancesMultipleIndices tests querying balances for multiple validator indices.
+func (s *BeaconKitE2ESuite) TestValidatorBalancesMultipleIndices() {
+	client := s.initBeaconTest()
+	indices := []phase0.ValidatorIndex{0, 1, 2}
+
+	balancesResp, err := client.ValidatorBalances(
+		s.Ctx(),
+		&beaconapi.ValidatorBalancesOpts{
+			State:   utils.StateIDHead,
+			Indices: indices,
+		},
+	)
+
+	s.Require().NoError(err)
+	s.Require().NotNil(balancesResp)
+	s.Require().Len(balancesResp.Data, len(indices))
+
+	// Verify all requested indices are present
+	returnedIndices := make(map[phase0.ValidatorIndex]bool)
```

Comment:
nit: consider `map[phase0.ValidatorIndex]struct{}` instead of `map[phase0.ValidatorIndex]bool`. The former is the idiomatic way to specify sets in go

Reply:
addressed